EssentialCSharp.Web/Services/Referrals/ReferralService.cs

```diff
@@ -0,0 +1,117 @@
+using System.Security.Claims;
+using EssentialCSharp.Web.Areas.Identity.Data;
+using EssentialCSharp.Web.Data;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.EntityFrameworkCore;
+using Sqids;
+
+namespace EssentialCSharp.Web.Services.Referrals;
+
+public class ReferralService(EssentialCSharpWebContext dbContext, SqidsEncoder<int> sqids, UserManager<EssentialCSharpWebUser> userManager) : IReferralService
+{
+    public async Task<string?> GetReferralIdAsync(string userId)
+    {
+        EssentialCSharpWebUser? user = await userManager.FindByIdAsync(userId);
+        return await GetReferralIdAsync(user);
+    }
+
+    public async Task<string?> GetReferralIdAsync(EssentialCSharpWebUser? user)
+    {
+        if (user is null)
+        {
+            return null;
+        }
+        else
+        {
+            // Check if the user already has a referrer ID
+            if (!string.IsNullOrEmpty(user.ReferrerId))
+            {
+                return user.ReferrerId;
+            }
+            else
+            {
+                Random random = Random.Shared;
+                string referrerId = sqids.Encode(random.Next());
+                user.ReferrerId = referrerId;
+
+                await userManager.AddClaimAsync(user, new Claim("ReferrerId", referrerId));
```

Comment:
Does this need to be stored in two different places in the database? (once in the new column on AspNetUsers, and once in AspNetUserClaims)?

Reply:
Good point, I am moving it just to the claims as that seems to make most sense for this unless you have an opposing opinion. (https://github.com/IntelliTect/EssentialCSharp.Web/pull/722/commits/b1cacc7693195f5733c0b25132869c4dce9459e6)